### ISSUE TITLE
Axis on example violin plot in docs is unreadable

### ISSUE THREAD

**Issue by brl0** [holoviz/holoviews#4887]
The xaxis in this [example](https://holoviews.org/gallery/demos/bokeh/life_expectancy_split_violin.html) is unreadable.

<details><summary>Expand for code from docs:</summary>

```python
import holoviews as hv
import pandas as pd

from holoviews import dim
from bokeh.sampledata.gapminder import regions, life_expectancy

hv.extension('bokeh')

group1 = 'South Asia'
group2 = 'East Asia & Pacific'

group1_countries = list(regions[regions.Group==group1].index)
group2_countries = list(regions[regions.Group==group2].index)

group1_df = life_expectancy.loc[group1_countries]
group1_df['Region'] = group1
group2_df = life_expectancy.loc[group2_countries]
group2_df['Region'] = group2

data = pd.concat([group1_df, group2_df]).reset_index().melt('Region', life_expectancy.columns, var_name='Year', value_name='Life Expectancy')

violin = hv.Violin(data, ['Year', 'Region'])

violin.opts(split='Region', responsive=True, min_height=500, show_legend=True, violin_width=1.5, legend_position='bottom_right', title='Life Expectancy by Year for Asian subregions', fontscale=1.5)
```
</details>

![image](https://user-images.githubusercontent.com/42480289/114251451-b1955580-9966-11eb-9f1e-a5785d269743.png)

I tried to fix it by using the `xticks` parameter, but that replaces the year with index values as seen here:
```python
violin.opts(split='Region', responsive=True, min_height=500, show_legend=True, violin_width=1.5, legend_position='bottom_right', title='Life Expectancy by Year for Asian subregions', fontscale=1.5, xticks=10, toolbar=None)
```
![image](https://user-images.githubusercontent.com/42480289/114251685-8f500780-9967-11eb-8c8d-e7940d84cca7.png)

Incidentally, if I then remove the xticks parameter from the line above and run it again, the xaxis is still changed, still showing index numbers exactly the same as the image above.  
```python
violin.opts(split='Region', responsive=True, min_height=500, show_legend=True, violin_width=1.5, legend_position='bottom_right', title='Life Expectancy by Year for Asian subregions', fontscale=1.5, toolbar=None)
```
So the opts method seems to be changing the plot object permanently, which seems contrary to my understanding of the way options typically work in holoviews.

So to summarize:

1. The example plot should probably be updated to look better.  I would be happy to submit a PR to do so, once I figure out how to actually do it, and guidance or suggestions would be appreciated.
2. The xticks option seems broken, although it could just be my understanding or expectation that is broken, and I would appreciate any clarity.
3. The fact that the opts changes the object permanently seems like another possible bug, but may again be a lack of understanding.

I would be happy to create separate issues if preferred.  Thanks in advance for any input.

**Comment by jbednar** [comment]
In this case a PR to add `xrotation=90` would fix it.

It should also be possible to skip labels, but there are some limitations on that for categorical and other special axes, so I'm not sure what's supported here.

In any case, `.opts` mutates the object it's used on; you can use `.options`  or `.opts(clone=True, ...)` if you want the functional equivalent. `.options` is the older approach, but it confused a lot of people and let to a bunch of code like `violin = violin.options(...)` that makes things harder to follow. Use whichever approach fits your mental model better!

**Comment by jbednar** [comment]
I went ahead and made a PR for that given that a release is imminent, so that it will fix the example on the website.
